test/e2e/job_attachment_bundle/dep_data_flow/windows_bundle/template.yaml

```
@@ -0,0 +1,288 @@
+          # When running all the tasks of a step, the input is not cleared
+          # between each task, so inputs created from other tasks may be included.
+          $INPUT_DIR_COUNT=(Get-ChildItem -Path {{Param.InputDir}}).Count
+          if (($INPUT_DIR_COUNT -lt 2) -or ($INPUT_DIR_COUNT -gt 5)) {
+            echo Starting dir contents is incorrect:
+            ls {{Param.InputDir}}
+            exit 1
+          }
+
+          # This file should be visible in later steps
+          cd {{Param.DataDir}}
+          Copy-Item "Step1.out" -Destination "{{Param.DataDir}}\$STEP.out"
+          echo "3.{{Task.Param.Frame}} Processed in $STEP" | Out-File -FilePath "$STEP.out" -Append -Encoding utf8
+
+          # This file should not be visible in later steps
+          cd {{Param.InputDir}}
+          Copy-Item "create_job_in.txt" -Destination "{{Param.InputDir}}\$STEP.out"
+          echo "2. Processed in $STEP" | Out-File -FilePath "$STEP.out" -Append -Encoding utf8
+
+          echo "Finished $STEP!"
+- name: Step1-2-3
+  hostRequirements:
+    attributes:
+    - name: attr.worker.os.family
+      anyOf:
+      - windows
+  dependencies:
+  - dependsOn: Step1-2
+  script:
+    actions:
+      onRun:
+        command: powershell
+        args: ["-File", "{{Task.File.step3}}"]
+    embeddedFiles:
+      - name: step3
+        filename: step3.ps1
+        runnable: true
+        type: TEXT
+        data: |
+          $STEP="Step1-2-3"
+
+          echo "Running $STEP!"
+
+          # DataDir should now include the output of every task in Step2, but not Step1.
+          echo "Checking DataDir: {{RawParam.DataDir}} / {{Param.DataDir}}"
+          if ((Get-ChildItem -Path {{Param.DataDir}}).Count -ne 6) {
+            echo "Starting dir contents is incorrect:"
+            Get-ChildItem -Path {{Param.DataDir}}
+            exit 1
+          }
+          # InputDir should not include the output of Step1 or Step2
+          echo "Checking InputDir: {{RawParam.InputDir}} / {{Param.InputDir}}"
+          if ((Get-ChildItem -Path {{Param.InputDir}}).Count -ne 2) {
+            echo "Starting dir contents is incorrect:"
+            Get-ChildItem -Path {{Param.InputDir}}
+            exit 1
+          }
+
+          # This file should be visible in later steps
+          cd {{Param.DataDir}}
+          Copy-Item "Step1-2.8.out" -Destination "{{Param.DataDir}}\$STEP.out"
+          Get-Content Step1-2.9.out | Out-File -FilePath "$STEP.out" -Append -Encoding utf8
+          Get-Content Step1-2.10.out | Out-File -FilePath "$STEP.out" -Append -Encoding utf8
+          Get-Content Step1-2.11.out | Out-File -FilePath "$STEP.out" -Append -Encoding utf8
+          echo "4. Processed in $STEP" | Out-File -FilePath "$STEP.out" -Append -Encoding utf8
+
+          # This file should not be visible in later steps
+          cd {{Param.InputDir}}
+          Copy-Item "create_job_in.txt" -Destination "{{Param.InputDir}}\$STEP.out"
```

nit: we're already in `Param.InputDir`. This is fine as long as InputDir is absolute path though
```suggestion
          cd {{Param.InputDir}}
          Copy-Item "create_job_in.txt" -Destination "$STEP.out"
```